python/cuspatial/cuspatial/io/geoseries_reader.py

```diff
@@ -0,0 +1,298 @@
+                )
+                for poly in geometry:
+                    current = offsets["polygons"]["polygons"][-1]
+                    num_rings = 1
+                    rings_current = offsets["polygons"]["rings"][-1]
+                    offsets["polygons"]["rings"].append(
+                        len(poly.exterior.coords) * 2 + rings_current
+                    )
+                    for interior in poly.interiors:
+                        rings_current = offsets["polygons"]["rings"][-1]
+                        offsets["polygons"]["rings"].append(
+                            len(interior.coords) * 2 + rings_current
+                        )
+                        num_rings = num_rings + 1
+                    offsets["polygons"]["polygons"].append(num_rings + current)
+                offsets["polygons"]["mpolys"].append(
+                    len(offsets["polygons"]["polygons"]) - 1
+                )
+        return offsets
+
+    def _read_geometries(self, geoseries, offsets):
+        """
+        Creates a set of buffers sized to fit all of the geometries and
+        iteratively populates them with geometry coordinate values.
+
+        Parameters
+        ----------
+        geoseries : A GeoPandas GeoSeries object.
+        offsets : The set of offsets that correspond to the geoseries argument.
+        """
+        buffers = {
+            "points": cudf.Series(
+                column_empty(offsets["points"][-1], dtype="float64")
+            ),
+            "multipoints": cudf.Series(
+                column_empty(offsets["multipoints"][-1], dtype="float64")
+            ),
+            "lines": cudf.Series(
+                column_empty(offsets["lines"][-1], dtype="float64")
+            ),
+            "polygons": {
+                "polygons": cudf.Series(
+                    column_empty(
+                        len(offsets["polygons"]["polygons"]), dtype="float64"
+                    )
+                ),
+                "rings": cudf.Series(
+                    column_empty(
+                        len(offsets["polygons"]["rings"]), dtype="float64"
+                    )
+                ),
+                "coords": cudf.Series(
+                    column_empty(
+                        offsets["polygons"]["rings"][-1], dtype="float64"
+                    )
+                ),
+            },
+        }
+        read_count = {
+            "points": 0,
+            "multipoints": 0,
+            "lines": 0,
+            "polygons": 0,
+        }
+        inputs = []
+        input_types = []
+        input_lengths = []
+        for geometry in geoseries:
+            if isinstance(geometry, Point):
+                # write a point to the points buffer
+                # increase read_count of points pass
+                i = read_count["points"] * 2
+                buffers["points"][i] = geometry.x
+                buffers["points"][i + 1] = geometry.y
```

Comment:
These are tiny host --> device copies that will be extremely slow. Instead of copying from host --> device in this loop, we should create `buffers` as host buffers and fill them in via this loop, then once they're filled in we can do bulk copies at the end to move them to the device.

Reply:
:+1: